inst/templates/tidy_contributing.md

```diff
@@ -0,0 +1,192 @@
+Guide](http://style.tidyverse.org) and use the
+[lintr](https://cran.r-project.org/package=lintr) package to find code
+which does not adhere to the style guide.
+
+``` r
+# install.packages("lintr")
+lintr::lint_package()
+```
+
+Remember to include style changes only in code you are contributing.
+
+### Documentation
+
+We use [roxygen2](https://cran.r-project.org/package=roxygen2),
+specifically with the [Markdown
+syntax](https://cran.r-project.org/web/packages/roxygen2/vignettes/markdown.html),
+to create `NAMESPACE` and all `.Rd` files. All edits to documentation
+should be done in roxygen comments above the associated function or
+object.
+
+``` r
+# install.packages("devtools")
+devtools::document()
+```
+
+See the `RoxygenNote` in [DESCRIPTION](DESCRIPTION) for the version of
+roxygen2 being used. It’s nice if a pull request includes updated
+documentation, but a good reason to NOT `document()` is if you have a
+different version of roxygen2 installed and that sprays minor formatting
+changes across `.Rd` files that have nothing to do with the PR.
+
+### Testing
+
+We use [testthat](https://cran.r-project.org/package=testthat).
+Contributions with test cases are easier to accept because the tests
+ensure the code does what it intends to do and nothing else. Without
+tests the maintainer needs to check the new functionality by hand, a
+burden you can lessen or remove by including tests. If you are not sure
+what parts of your code is covered by tests
+[covr](https://cran.r-project.org/package=covr) is a great tool to use
+before submission. Just run the following to get a local coverage report
+of the package so you can see exactly what lines are not covered in the
+project.
+
+``` r
+# install.packages("covr")
+co <- covr::package_coverage()
+covr::report(co)
+```
+
+Before submitting your changes, make sure that the package either still
+passes `R CMD check`, or that the warnings and/or notes have not changed
+as a result of your edits.
+
+### The Pull Request
+
+Your pull request should clearly and concisely state the motivating the
+need for change. Each change and each PR) should correspond to a branch.
+The best way to check exactly what changes you are proposing is to use
+`git diff` prior to submitting your contribution. This will ensure it
+contains only the changes necessary for the new functionality.
+
+If the PR is related to an issue, link to it in the description, with
+[the `#15`
+syntax](https://help.github.com/articles/autolinked-references-and-urls/)
+and the issue slug for context. If the PR is meant to close an issue,
+make sure one of the commit messages includes [text like `closes #44` or
+`fixes
+#101`](https://help.github.com/articles/closing-issues-using-keywords/).
+Provide the issue number and slug in the description, even if the issue
+is mentioned in the title, because auto-linking does not work in the PR
+title.
+
+  - GOOD PR title: *“Obtain user’s intent via mind-reading; fixes
```

Comment:
Should never put "fixes" in title (house style)

Reply:
👌 should I submit a PR this for googledrive too
https://github.com/tidyverse/googledrive/blame/master/CONTRIBUTING.md#L23
or is there a cascade of specificity when it comes to style? (i.e. googledrive package-wide style can override tidyverse verse-wide style)

Reply:
I think once we've got this solid, you'll have to PR to every tidyverse package :)

Reply:
I'm trying to clarify this elsewhere.